src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+import UploadFileIcon from '@mui/icons-material/UploadFile';
+import FilePresentIcon from '@mui/icons-material/FilePresent';
+import * as React from 'react';
+import { type DropzoneOptions, useDropzone } from 'react-dropzone';
+import {
+	Box,
+	Link,
+	Typography,
+	CircularProgress,
+	Avatar,
+	InputLabel,
+	List,
+	ListItem,
+	ListItemText,
+} from '@mui/material';
+import type { WidgetProps } from '@rjsf/utils';
+
+export interface OnFileReadSuccessParams {
+	dataUrl: string;
+	uploadedFile: File;
+}
+
+type FileWidgetProps = WidgetProps & {
+	onChange: (params?: OnFileReadSuccessParams) => void;
+	onError?: (errors: string[]) => void;
+	value?: OnFileReadSuccessParams | string;
+};
+
+export const FileWidget = ({
+	onChange,
+	onError,
+	value,
+	...props
+}: FileWidgetProps) => {
+	const [isLoading, setIsLoading] = React.useState<boolean>(false);
+	const [loadingPercentage, setLoadingPercentage] = React.useState<number>(0);
+	const [errors, setErrors] = React.useState<string[]>([]);
+
+	const accept =
+		props.uiSchema?.['ui:options']?.accept ??
```

Comment:
We should destructure all these `props.`s in the parameter's list instead of using `props.`